earth_enterprise/src/fusion/autoingest/sysman/StateUpdater_unittest.cpp

```diff
@@ -977,6 +985,68 @@ TEST_F(StateUpdaterTest, Cancel) {
   assertStateNotSet(sm, "ci3");
 }
 
+TEST_F(StateUpdaterTest, FailedInput) {
+  SetVersions(sm, {
+                    MockVersion("a"),
+                    MockVersion("b")
+                  });
+  SetListenerInput(sm, "a", "b");
+
+  GetMutableVersion(sm, "a")->state = AssetDefs::Waiting;
+  GetMutableVersion(sm, "b")->state = AssetDefs::InProgress;
+
+  // MockVersion::InputStatesAffectMyState will return true, so setting this input
+  // to a Failed state should cause the listener to become Blocked
+  updater.SetAndPropagateState(fix("b"), AssetDefs::Failed, [](AssetDefs::State) { return true; });
+
+  assertStateSet(sm, "a");
+  assertStateSet(sm, "b");
+  assert(AssetDefs::Blocked == GetMutableVersion(sm, "a")->state);
+  assert(AssetDefs::Failed == GetMutableVersion(sm, "b")->state);
+}
+
+TEST_F(StateUpdaterTest, FailedInputDontCare) {
+  SetVersions(sm, {
+                    MockVersion("a"),
+                    MockVersion("b")
+                  });
+  SetListenerInput(sm, "a", "b");
+
+  GetMutableVersion(sm, "a")->state = AssetDefs::Waiting;
+  GetMutableVersion(sm, "b")->state = AssetDefs::InProgress;
+
+  GetMutableVersion(sm, "a")->inputStatesAffectMyState = false;
+
+  // MockVersion::InputStatesAffectMyState will return true, so setting this input
+  // to a Failed state should cause the listener to become Blocked
```

Comment:
Since we have set **inputStatesAffectMyState = false**, should this comment not state that the listener should not become Blocked?

Reply:
Good catch. Copy/paste strikes again ;)